zebra/zebra_wrap_script.c

```diff
@@ -0,0 +1,290 @@
+		l = search_current_word(current_str, k,
+					current_word);
+		if (l < 0)
+			break;
+		k += l;
+		/* no json obj. fields are filled in
+		 */
+		if (json_obj == NULL) {
+			if (zebra_wrap_debug
+			    & SCRIPT_ITEM_LIST)
+				zlog_err("SCRIPT: (%d)ITEM %s",
+					 nb_items, current_word);
+			item[nb_items].name =
+				XSTRDUP(MTYPE_TMP, current_word);
+		} else {
+			/* if a field has no column, create "misc"
+			 * column
+			 */
+			if (!item[nb_items].name) {
+				item[nb_items].name =
+					XSTRDUP(MTYPE_TMP, "misc");
+				keep_item = true;
+				item[nb_items].attribute =
+					XSTRDUP(MTYPE_TMP,
+						current_word);
+			} else if (item[nb_items].attribute) {
+				/* store last elements in attribute
+				 */
+				char temp_word[DATA_LINE_MAX];
+
+				snprintf(temp_word,
+					 DATA_LINE_MAX,
+					 "%s %s",
+					 item[nb_items].attribute,
+					 current_word);
+				XFREE(MTYPE_TMP,
+				      item[nb_items].attribute);
+				item[nb_items].attribute =
+					XSTRDUP(MTYPE_TMP,
+						temp_word);
+			}
+			if (!keep_item) {
+				json_object_string_add(json_obj,
+						       item[nb_items].name,
+						       current_word);
+				if (zebra_wrap_debug
+				    & SCRIPT_ELEMENT_LIST)
+					zlog_err("(%d)ITEM Obtained "
+						 "for %s is %s",
+						 nb_items,
+						 item[nb_items].name,
+						 current_word);
+			}
+		}
+		if (!keep_item)
+			nb_items++;
+		if (nb_items >= ITEM_MAXIMUM) {
+			int m;
+
+			for (m = 0; m < ITEM_MAXIMUM; m++)
+				XFREE(MTYPE_TMP, item[m].name);
+			if (json_obj)
+				json_object_free(json_obj);
+			return -1;
+		}
+	}
+	/* store last attribute to json
+	 */
+	if (keep_item) {
+		json_object_string_add(json_obj,
+				       item[nb_items].name,
+				       item[nb_items].attribute);
+		if (zebra_wrap_debug & SCRIPT_ITEM_LIST)
+			zlog_err("(%d)ITEM Obtained for %s is %s",
+				 nb_items,item[nb_items].name,
+				 item[nb_items].attribute);
+		XFREE(MTYPE_TMP, item[nb_items].attribute);
+		item[nb_items].attribute = NULL;
+		XFREE(MTYPE_TMP, item[nb_items].name);
+		item[nb_items].name = NULL;
+	}
+	return 0;
+}
+
+/* script : command line to execute in a shell script
+ * return_data : set to true if want to get back some information
+ * begin_at_line : the line number where to begin parsing headers and other
+ * - ex: following dump example begins at line 2, where header is located
+ *    # iptables -t mangle -L PREROUTING -v
+ *    Chain PREROUTING (policy ACCEPT 150k packets, 7426 bytes)     (## line 0)
+ *     pkts bytes target    prot opt in    out source   destination (## line 1)
+ *     0     0     DROP      all --  any   any anywhere  anywhere
+ *          match-set match0x55f44       (## line 2)
+ * json_obj_list : the json structure mapped to the output, ranked with line nb
+ * - ex: above dump gives following
+ * { "2":{"pkts":"0","bytes":"0","target":"MARK","prot":"all", \
+ *           "opt":"--","in":"any",..}}
+ */
+
+int zebra_wrap_script(char *script, bool return_data,
```

Comment:
I know "const" is under used in FRR, but whevenever possible, it should. For instance, here "char *script" is clearly a const.